neo/Consensus/ConsensusContext.cs

```diff
@@ -206,18 +206,59 @@ private void SignPayload(ConsensusPayload payload)
             payload.Witness = sc.GetWitnesses()[0];
         }
 
+        /// <summary>
+        /// Prevent that block exceed the max size
+        /// </summary>
+        /// <param name="txs">Ordered transactions</param>
+        internal void EnsureMaxBlockSize(IEnumerable<Transaction> txs)
+        {
+            var transactions = txs.ToList();
+            // Limit Speaker proposal to the limit `MaxTransactionsPerBlock` or all available transactions of the mempool
+            TransactionHashes = new UInt256[Math.Min(transactions.Count, NativeContract.Policy.GetMaxTransactionsPerBlock(Snapshot))];
+            
+            Block.Transactions = new Transaction[0];
+            // TODO: Real expected witness
```

Comment:
Shouldn't we fix this before merging?

Reply:
Yes

Reply:
@igormcoelho could you help me with this? we need to know the expected witness size